### ISSUE TITLE
Hint diagnostics don’t show up in Problems window

### ISSUE THREAD

**Issue by arzg** [rust-analyzer/rust-analyzer#7436]
The recently introduced ‘replace `.filter_map().next()` with `.find_map()`’ diagnostic doesn’t show up in VS Code’s Problems window. Although Clippy does catch this and will make it appear in Problems, If I understand correctly, the eventual plan is for all diagnostics to be on-the-fly in rust-analyzer. When this happens, it would be desirable to have diagnostics in the Problems window so you can be sure that they’ve all been fixed.

**Comment by bjorn3** [comment]
It is vscode which decides what is shown in the problem pane, not rust-analyzer. Vscode may have a config option for it. Showing hints in the problems pane would be quirete noisy though as several non-problems are given the hint diagnostic level like code blocks being disabled due to `#[cfg]`.

**Comment by arzg** [comment]
Ok, interesting. Could hints that actually are problems be promoted to warnings?

**Comment by bjorn3** [comment]
`.filter_map().next()` isn't a problem. It is just a lint.

**Comment by arzg** [comment]
True, but I think it would be helpful to see lints found across an entire codebase in one place (whether that’s the Problems window or not) to make sure I’ve seen each one. Is there a place more appropriate for this than Problems?

Also, do you think it’s better to hold off from doing this for now because the lints have too many false positives? I personally haven’t had any, but then again the code I work with doesn’t have too many macros or generics.

**Comment by flodiebold** [comment]
I think Hint diagnostics not showing up in the problems window is quite intentional, maybe even the point of that severity level. Maybe we could provide a setting to change the severity of diagnostics in the future.
